src/gleam/erlang/file.gleam

```diff
@@ -0,0 +1,169 @@
+/// - `Enametoolong` :: Filename too long.
+/// - `Enfile` :: File table overflow
+/// - `Enobufs` :: No buffer space available.
+/// - `Enodev` :: No such device.
+/// - `Enolck` :: No locks available.
+/// - `Enolink` :: Link has been severed.
+/// - `Enoent` :: No such file or directory.
+/// - `Enomem` :: Not enough memory.
+/// - `Enospc` :: No space left on device.
+/// - `Enosr` :: No STREAM resources.
+/// - `Enostr` :: Not a STREAM.
+/// - `Enosys` :: Function not implemented.
+/// - `Enotblk` :: Block device required.
+/// - `Enotdir` :: Not a directory.
+/// - `Enotsup` :: Operation not supported.
+/// - `Enxio` :: No such device or address.
+/// - `Eopnotsupp` :: Operation not supported on socket.
+/// - `Eoverflow` :: Value too large to be stored in data type.
+/// - `Eperm` :: Not owner.
+/// - `Epipe` :: Broken pipe.
+/// - `Erange` :: Result too large.
+/// - `Erofs` :: Read-only file system.
+/// - `Espipe` :: Invalid seek.
+/// - `Esrch` :: No such process.
+/// - `Estale` :: Stale remote file handle.
+/// - `Etxtbsy` :: Text file busy.
+/// - `Exdev` :: Cross-domain link.
+///
+/// ## Erlang Errors
+/// - `Badarg` :: Bad argument; usually the argument is of an invalid type. Due
+///  to Gleam's strong typing, this error should not be encountered when the
+///  functions of this module are called from Gleam code.
+/// - `Terminated` :: File server process is terminated.
+/// - `SystemLimit` :: A system limit was hit, probably not enough ports.
+pub type Reason {
+  Eacces
+  Eagain
+  Ebadf
+  Ebadmsg
+  Ebusy
+  Edeadlk
+  Edeadlock
+  Edquot
+  Eexist
+  Efault
+  Efbig
+  Eftype
+  Eintr
+  Einval
+  Eio
+  Eisdir
+  Eloop
+  Emfile
+  Emlink
+  Emultihop
+  Enametoolong
+  Enfile
+  Enobufs
+  Enodev
+  Enolck
+  Enolink
+  Enoent
+  Enomem
+  Enospc
+  Enosr
+  Enostr
+  Enosys
+  Enotblk
+  Enotdir
+  Enotsup
+  Enxio
+  Eopnotsupp
+  Eoverflow
+  Eperm
+  Epipe
+  Erange
+  Erofs
+  Espipe
+  Esrch
+  Estale
+  Etxtbsy
+  Exdev
+  Bardarg
+  Terminated
+  SystemLimit
+}
+
+/// Read the contents of the given file.
+///
+/// Returns a Result containing the file's contents as a String if the
+/// operation was successful, or Reason if the operation failed.
+///
+/// ## Examples
+///
+///    > read_file("example.txt")
+///    Ok("Hello, World!")
+///
+///    > read_file("does_not_exist.txt")
+///    Error(Enoent)
+///
+pub external fn read_file(filename: String) -> Result(BitString, Reason) =
+  "gleam_erlang_ffi" "read_file"
+
+/// Write the given contents to a file of the given name.
+///
+/// Returns a Result with Nil if the operation was successful or a Reason
+/// otherwise.
+///
+/// ## Examples
+///
+///    > write_file("file.txt", "Hello, World!")
+///    Ok(Nil)
+///
+///    > write_file("does_not_exist/file.txt", "Hello, World!")
+///    Error(Enoent)
+///
+pub external fn write_file(
+  filename: String,
+  contents: BitString,
+) -> Result(Nil, Reason) =
+  "gleam_erlang_ffi" "write_file"
```

Comment:
BIkeshedding on names, how about this?

```suggestion
pub external fn write_bits(
  to: String,
  contents: BitString,
) -> Result(Nil, Reason) =
  "gleam_erlang_ffi" "write_file"
```

Reply:
Ooh, I do like `to:`. Makes me feel like we should also have the argument be labelled `from:` in `read_bits`.